src/scanner.rs

```diff
@@ -42,6 +57,12 @@ impl ScanConfig {
         self
     }
 
+    #[inline]
+    pub fn extend_filters(mut self, filters: Vec<Filter>) -> Self {
+        self.filters.extend(filters);
+        self
+    }
+
     /// Filter scanned devices based on the device address
     #[inline]
     pub fn filter_by_address(mut self, func: impl Fn(BDAddr) -> bool + Send + Sync + 'static) -> Self {
```

Comment:
Should these `filter_by` methods be removed, since you can now use the `Filter`?

Reply:
I think the user can customize the filter behavior, if not then use the default behavior

Reply:
Maybe `Filter::Address` can use a string